src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1448,30 +1446,29 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         auto update{ winrt::make<ScrollPositionChangedArgs>(viewTop,
                                                             viewHeight,
                                                             bufferSize) };
-        if (!_inUnitTests && _updateScrollBar)
-        {
-            _updateScrollBar->Run(update);
-        }
-        else
+
+        if (_inUnitTests) [[unlikely]]
         {
             _ScrollPositionChangedHandlers(*this, update);

```

Comment:
Here's the incorrect if condition. It would enter this unit-test branch as soon as we `_updateScrollBar.reset()`.